Documentation/ceph-disaster-recovery.md

```diff
@@ -237,6 +238,109 @@ kubectl -n rook-ceph scale deployment rook-ceph-operator --replicas=1
+ ]'
+
+kubectl -n rook-ceph-tools delete pod -l app=rook-ceph-tools
+```
+
+### Collect monmap from healthy osds
+
+```console
+OSD_IDS=<healthy osd's ids like "0 1 2">
+TMP_MON_STORE=<temporal monstore's file>
+
+for i in ${OSD_IDS}; do
+    name=$(kubectl get -n rook-ceph pod -l app=rook-ceph-osd -l osd="${i}" -o json | jq -r '.items[].metadata.name')
+    kubectl exec -i -n rook-ceph ${name} -- rm -rf "${TMP_MON_STORE}"
+    kubectl exec -i -n rook-ceph ${name} -- mkdir -p "${TMP_MON_STORE}"
+    tar cf - -C "${TMP_MON_STORE}" . | kubectl exec -i -n "rook-ceph "${name}" -- tar xf - -C "${TMP_MON_STORE}"
+    kubectl exec -i -n rook-ceph ${name} -- ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-"${i}" --no-mon-config --op update-mon-db --mon-store-path "${TMP_MON_STORE}"
+    rm -rf "${TMP_MON_STORE}"/*
+    kubectl exec -i -n rook-ceph "${name}" -- tar cf - -C "${TMP_MON_STORE}" . | tar xf - -C "${TMP_MON_STORE}"
+done
+```
+
+### Rebuild monstore
+
+```console
+FSID=$(kubectl -n rook-ceph get secret rook-ceph-mon -o jsonpath='{.data.fsid}' | base64 -d)
+kubectl exec -i -n rook-ceph deployment/rook-ceph-tools -- rm -rf "${TMP_MON_STORE}"
+kubectl exec -i -n rook-ceph deployment/rook-ceph-tools -- mkdir -p "${TMP_MON_STORE}"
+tar cf - -C "${TMP_MON_STORE}" . | kubectl exec -i -n rook-ceph deployment/rook-ceph-tools -- tar xf - -C "${TMP_MON_STORE}"
+kubectl exec -i -n rook-ceph deployment/rook-ceph-tools -- ceph-monstore-tool --fsid="${FSID}" "${TMP_MON_STORE}" rebuild -- --keyring /keyrings/keyring --mon-ids ${MON_IDS}
+rm -rf "${TMP_MON_STORE}"
+mkdir -p "${TMP_MON_STORE}"
+kubectl exec -i -n rook-ceph deployment/rook-ceph-tools -- tar cf - -C "${TMP_MON_STORE}" . | tar xf - -C "${TMP_MON_STORE}"
+```
+
+### Replace the corrupted monstore with the recovered copy.
+
+```console
+MON_IDS=<mon's ids, like "a b c">
+
+for i in ${MON_IDS}; do
+    name=$(kubectl get -n rook-ceph pod -l app=rook-ceph-mon -l mon="${i}" -o json | jq -r '.items[].metadata.name')
+    STORE_DB=/var/lib/ceph/mon/ceph-"${i}"/store.db
+    kubectl exec -i -n rook-ceph "${name}" -- rm -rf "${STORE_DB}".corrupted
+    kubectl exec -i -n rook-ceph "${name}" -- mkdir -p "${STORE_DB}"
+    kubectl exec -i -n rook-ceph "${name}" -- mv "${STORE_DB}" "${STORE_DB}".corrupted
+    kubectl exec -i -n rook-ceph "${name}" -- mkdir -p "${STORE_DB}"
+    tar cf - -C "${TMP_MON_STORE}"/store.db . | kubectl exec -i -n rook-ceph "${name}" -- tar xf - -C "${STORE_DB}"
+    kubectl exec -i -n rook-ceph "${name}" -- chown -R ceph:ceph "${STORE_DB}"
+done
+```
+
+### Aftertreatment
+
+* Restore deployments of mons, osds, and toolbox.
+* Restart the cluster
+
+```console
+kubectl exec -i -n rook-ceph deployment/rook-ceph-tools -- ceph mon --connect-timeout 60 enable-msgr2
+kubectl -n rook-ceph scale deployment rook-ceph-operator --replicas=1
+```
+
+* Restart operator by `kubectl -n rook-ceph delete pod -l app=rook-ceph-operator`
```

Comment:
Why restart the operator right after it was just scaled up? Maybe we don't need this step?